### PR TITLE
Add Xcode 12.5 swift test job

### LINKED ISSUE #3048: Add Xcode 12.5 to our CI Builds
CircleCI added support for Xcode 12.5. This Xcode version is known for breaking some project builds. We can make the following changes:
1. Move jobs that use Xcode 12.0 to run only on "release-v2.0" and "main" branches. (Motivation: I think Xcode 12.0 isn't popular, but at least we will spot the regression after merges). 
2. Add Xcode 12.5 build job as a replacement for Xcode 12.0 jobs. 

@mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by S2Ler**
Fixes #3048

- Move Xcode 12.4 swift test to a new "extended-workflow". 
- Create Xcode 12.5 swift test.
- Move iOS 13.7 Xcode 12.5 swift test to the "extended-workflow".
- "extended-workflow" should run an extended set of tests once a day that will
allow us to spot regressions in more cases while saving the cost on CI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -296,7 +296,27 @@ jobs:
...
+          iOS: "14.4"          
+          device: "iPhone 12 Pro Max"     
+      - spm-test-job:
+          name: "swift test; Xcode 12.4; iOS 13.7"
+          xcode: "12.4.0"
+          iOS: "13.7"          
+          device: "iPhone 11 Pro Max"
+    triggers:
+       - schedule:
+           cron: "0 0 * * *"
```
Comment (MaximAlien) [inline .circleci/config.yml:313]: Would probably make sense to add informational comment describing how often this job will be executed.

Comment (S2Ler) [reply .circleci/config.yml:313]: Can't agree more, looks cryptic

File: Package.swift [hunk 2]
```diff
@@ -61,7 +61,10 @@ let package = Package(
             resources: [.copy("MBXInfo.plist")]),
         .target(
             name: "CTestHelper",
-            dependencies: ["MapboxMobileEvents"]),
+            dependencies: [
+                "MapboxMobileEvents",
+                "MapboxCoreNavigation",
```
Comment (1ec5) [inline Package.swift:66]: Interesting that the build was passing without this change, but I agree with adding it as a dependency.

**Review by 1ec5 — APPROVED** [review]
Looks good. The cron comment would be nice too.

File: Package.swift (hunk 2 above)
Comment (S2Ler) [reply Package.swift:66]: Yeah, no idea why wouldn't it build...

File: .circleci/config.yml [hunk 3]
```diff
@@ -296,7 +296,27 @@ jobs:
...
+          iOS: "14.4"          
+          device: "iPhone 12 Pro Max"     
+      - spm-test-job:
+          name: "swift test; Xcode 12.4; iOS 13.7"
+          xcode: "12.4.0"
+          iOS: "13.7"          
+          device: "iPhone 11 Pro Max"
+    triggers:
+       - schedule:
+           cron: "0 0 * * *" # Once per day at 00:00
```
Comment (azarovalex) [inline .circleci/config.yml:313]: Is there any reason to run CI build every day? I don't think we will monitor these builds every day. IMHO CI builds for every PR should be enough.

Comment (S2Ler) [reply .circleci/config.yml:313]: Yes, the motivation is in the description. 

Visibility issue is a thing which can be solved with a slack bot integration.

**Pull request merged**